lib/src/bridge/characteristics_mixin.dart

```diff
@@ -47,16 +47,16 @@ mixin CharacteristicsMixin on FlutterBLE {
               ArgumentName.transactionId: transactionId
             },
           )
-          .catchError((errorJson) =>
-              Future.error(BleError.fromJson(jsonDecode(errorJson.details))))
+          .catchError((errorJson) async =>
+              throw BleError.fromJson(jsonDecode(errorJson.details))
+          )
           .then((rawValue) {
-            String rawJsonValue = "";
-            if (rawValue is String) {
-              rawJsonValue = rawValue;
+            if (rawValue == null) {
+              throw Exception("rawValue cannot be null");
```

Comment:
This exception doesn't give any value to the consumer of the API.

Reply:
yes, I know. I wanted to discuss what it should throw, but now it's postponed as I've reverted the commit. ;)